### DISCUSSION TITLE
Tips on setting up GitHub pages action

### DISCUSSION THREAD

**Discussion by vcavallo** [srid/emanote#507] [General]
# Some tips on Configuring the GitHub Pages Action

(From someone who just stumbled through it for hours)

## `GITHUB_TOKEN` Settings

Make sure your (implicitly created) `GITHUB_TOKEN` has the proper privileges in Repo Settings > Actions > General.

![image](https://github.com/srid/emanote/assets/3201135/ae76c917-fc77-4f10-959c-f74677920c55)

## Pages Settings

In Repo Settings > Pages, you _may_ need to make sure "Deploy from branch" is properly set (the `gh-pages` branch will be created after the emanote action runs at least once) and your custom CNAME domain is set (if using):

![image](https://github.com/srid/emanote/assets/3201135/61fcea5c-9517-47e7-b13d-ef6036cd84d5)

## Other

- in `.github/workflows/publish.yml` make sure your `cname` entry is set (if using a custom domain)
- set the `baseUrl` property correctly in `flake.nix` (see below for an example)
- if your site has broken links and you'd like to deploy it that way, adjust the `emanote` flags in in `flake.nix` to include `allowBrokenLinks`:
```nix
        emanote = {
          sites."default" = {
            layers = [ ./. ];
            layersString = [ "." ];
            baseUrl = "/"; 
            allowBrokenLinks = true; # HERE!
          };
```

**Comment by srid** [comment]
Nice! Would you like to open a PR adding a new page to the docs?

**Comment by srid** [comment]
Another way: https://github.com/srid/emanote-template/pull/81
